SConstruct

```diff
@@ -586,25 +609,138 @@ def list_sources(base, suffixes):
                     yield os.path.normpath(path)
     return list(_iter(base))
 
-def add_soci_sources(env, object_builder, is_unity):
+def append_sources (result, *filenames, **kwds):
+    result.append([filenames, kwds])
+
+def get_soci_sources(style):
+    result = []
     cpp_path = [
         'src/soci/src/core',
         'src/sqlite',]
-    object_builder.add_source_files(
+    append_sources(result,
         'src/ripple/unity/soci.cpp',
         CPPPATH = cpp_path)
-    if is_unity:
-        object_builder.add_source_files(
+    if 'unity' == style:
```

Comment:
I should say that value in C or C++ Yoda conditions _do_ have, but not in Python.